ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -23,3 +26,93 @@ def __init__(self, resource_name="", *args, **kwargs):
         super(StorageCluster, self).__init__(
             resource_name=resource_name, kind='StorageCluster', *args, **kwargs
         )
+
+
+def count_cluster_osd():
+    """
+    The function returns the number of cluster OSDs
+
+    Returns:
+         osd_count (int): number of OSD pods in current cluster
+
+    """
+    storage_cluster_obj = StorageCluster(
+        resource_name=config.ENV_DATA['storage_cluster_name'],
+        namespace=config.ENV_DATA['cluster_namespace'],
+    )
+    storage_cluster_obj.reload_data()
+    osd_count = (
+        int(storage_cluster_obj.data['spec']['storageDeviceSets'][0]['count'])
+        * int(storage_cluster_obj.data['spec']['storageDeviceSets'][0]['replica'])
+    )
+    return osd_count
+
+
+def add_capacity(capacity_string):
+    """
+   Add storage capacity to the cluster
+
+   Args:
+       capacity_string(int): Size of the storage to add
+   Returns:
+       True if capacity was added, False if not
+
+   """
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    old_osd = count_cluster_osd()
+    sc = get_storage_cluster()
+    old_deviceSet_count = sc['items'][0]['spec']['storageDeviceSets'][0]['count']
+    osd_size = parse_size_to_int(sc['items'][0]['spec']['storageDeviceSets'][0]['dataPVCTemplate']
+                                 ['spec']['resources']['requests']['storage'])
+    capacity_to_add = parse_size_to_int(capacity_string) / osd_size + old_deviceSet_count
+
+    # adding the storage capacity to the cluster
+    ocp.patch(
+        resource_name=sc['metadata']['name'],
+        params=f'[{{"op": "replace", "path": "/spec/storageDeviceSets/0/count", '
+               f'"value":{capacity_to_add}}}]'
+    )
+
+    # cluster health check
+    if not utils.ceph_health_check:
+        log.info("Capacity was not added")
+        return False
+
+    # osd amount validation
+    new_osd_count = count_cluster_osd()
+    expected = parse_size_to_int(capacity_string) / osd_size * 3 + old_osd
+    if not expected == new_osd_count:
+        log.info("Capacity was not added")
+        return False
+
+    log.info(f"{capacity_string} was added")
+    return True
+
+
+def get_storage_cluster(namespace=defaults.ROOK_CLUSTER_NAMESPACE):
+    """
+   Get storage cluster name
+
+   Args:
+       namespace (str): Namespace of the resource
+   Returns:
+       yaml: Storage cluster yaml
+   """
+
+    sc_obj = OCP(kind=constants.STORAGECLUSTER, namespace=namespace)
```

Comment:
why not initializing StorageCluster instance instead?

Reply:
function already existed, just changed it to return a dictionary instead of the storage cluster name. It uses a very good 'get' function from OCP

Reply:
was changed in the add capacity function. this one is not used